src/shims/unix/socket.rs

```diff
@@ -22,15 +31,62 @@ impl FileDescription for SocketPair {
     ) -> InterpResult<'tcx, io::Result<()>> {
         Ok(Ok(()))
     }
+
+    fn read<'tcx>(
```

Comment:
Don't we have to indicate EOF at some point in `read`? Like, when the other side got closed and all data has been read, or so?